File: src/com/mishiranu/dashchan/content/net/UserAgentProvider.java
package com.mishiranu.dashchan.content.net;

import android.app.Application;
import android.webkit.WebSettings;
import android.webkit.WebView;

import androidx.annotation.MainThread;

public class UserAgentProvider {
	private static final UserAgentProvider INSTANCE = new UserAgentProvider();

	private String userAgent;

	@MainThread
	public static void initialize(Application appContext) {
		INSTANCE.userAgent = WebSettings.getDefaultUserAgent(appContext);
	}

	public static UserAgentProvider getInstance() {
		if (INSTANCE.userAgent == null) {
			throw new IllegalStateException("UserAgentProvider is not initialized");
		}
		return INSTANCE;
	}

	private UserAgentProvider() {
	}

	public String getUserAgent() {
		return userAgent;
	}

}

File: src/com/mishiranu/dashchan/content/net/UserAgentProvider.kt
package com.mishiranu.dashchan.content.net

import android.app.Application
import android.webkit.WebSettings
import androidx.annotation.MainThread

class UserAgentProvider private constructor() {
    private var userAgent: String? = null

    fun getUserAgent(): String? = userAgent

    companion object {
        private val INSTANCE = UserAgentProvider()

        @JvmStatic
        @MainThread
        fun initialize(appContext: Application) {
            INSTANCE.userAgent = WebSettings.getDefaultUserAgent(appContext)
        }

        @JvmStatic
        fun getInstance(): UserAgentProvider {
            checkNotNull(INSTANCE.userAgent) { "UserAgentProvider is not initialized" }
            return INSTANCE
        }
    }
}
